src/core/wee-eval.h

```diff
@@ -42,6 +42,8 @@ enum t_eval_comparison
 {
     EVAL_COMPARE_REGEX_MATCHING = 0,
     EVAL_COMPARE_REGEX_NOT_MATCHING,
+    EVAL_COMPARE_MATCHING,
```

Comment:
For less ambiguity, I would use name `EVAL_COMPARE_STRING_MATCHING` and `EVAL_COMPARE_STRING_NOT_MATCHING`.